quill-cassandra-monix/src/main/scala/io/getquill/CassandraMonixContext.scala

```diff
@@ -37,22 +37,20 @@ class CassandraMonixContext[N <: NamingStrategy](
   override type RunQuerySingleResult[T] = T
   override type RunBatchActionResult = Unit
 
-  protected def page(rs: ResultSet): Task[Iterable[Row]] = Task.defer {
-    val available = rs.getAvailableWithoutFetching
-    val page = rs.asScala.take(available)
-
-    if (rs.isFullyFetched)
-      Task.now(page)
-    else
-      Task.fromFuture(rs.fetchMoreResults().asScalaWithDefaultGlobal).map(_ => page)
-  }
+  protected def page(rs: AsyncResultSet): Task[Iterable[Row]] =
+    Task.defer {
+      val page = rs.currentPage().asScala
+      if (rs.hasMorePages)
+        Task.from(rs.fetchNextPage().toCompletableFuture.toScala).map(_ => page)
+      else
+        Task.now(page)
+    }
 
   def streamQuery[T](fetchSize: Option[Int], cql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor)(info: ExecutionInfo, dc: DatasourceContext): Observable[T] = {
-
     Observable
       .fromTask(prepareRowAndLog(cql, prepare))
-      .mapEvalF(p => session.executeAsync(p).asScalaWithDefaultGlobal)
-      .flatMap(Observable.fromAsyncStateAction((rs: ResultSet) => page(rs).map((_, rs)))(_))
+      .mapEvalF(p => session.executeAsync(p).toScala)
```

Comment:
This runs to a CompletableFuture?

Reply:
`  default **CompletionStage**<AsyncResultSet> executeAsync(@NonNull Statement<?> statement) {
    return Objects.requireNonNull(
        execute(statement, Statement.ASYNC), "The CQL processor should never return a null result");
  }`